src/components/Filters/Filters.tsx

```diff
@@ -24,20 +24,22 @@ interface FilterProps<F, G extends Value = string> {
   };
   /** Specifies the layout of the filters. If not supplied it will use the default (horizontal) layout. Using the 'vertical' layout will also remove the "More Filters" button/modal */
   vertical?: boolean;
+  /** Specifies the number of in line filters before more filters modal  */
+  numberOfInlineFilters?: number;
 }
 
 function Filters<F, G extends Value = string>(props: FilterProps<F, G>) {
-  const { filter, onChange, filterDefs, groupBy, vertical = false } = props;
+  const { filter, onChange, filterDefs, groupBy, vertical = false, numberOfInlineFilters = 4 } = props;
   const testId = useTestIds(props, filterTestIdPrefix);
 
   const { openModal } = useModal();
-  const numberOfPageFilters = groupBy ? 2 : 3;
+  const numberOfPageFilters = numberOfInlineFilters - (groupBy ? 1 : 0);
```

Comment:
I'm not entirely sure, when I found it referenced/used it was always 1 less than without the group by

Reply:
@stephenh - looks like you may have touched this, the `-1` for groupBy sound about right to you?

Reply:
Ah yeah, was going to comment, but the rationale was:

* If the page does not have a group by, we have "4 spots" to put filters in
* If the page does have a group by, that takes one of the spots, so we have "3 spots" left for filters

I think something like:

```
const { groupBy, numberOfInlineFilters = groupBy ? 3 : 4 } = props
```

Would do this w/o the need for two `numberOfInlineFilters` and `numberOfPageFilters` variables.

Reply:
@stephenh 

@crshannon and I are discussing this now

Should we move the group by into this count/more filters? It seems intentional that it is not.

this issue originated out of responsive and the ability to see the right side of the job logs page header. There is not space there for buttons to be used 
Is the group by something we always want to see? Or would we like to expand this work in the future to make groupy/filter truly responsive. 

![Screen Shot 2022-12-07 at 12 04 04 PM](https://user-images.githubusercontent.com/100221471/206274185-8534603a-b7ce-4ca5-979d-48bcaae920b3.png)

Reply:
@aTrujilloHomeBound sorry, I'm not completely following...

> Should we move the group by into this count/more filters? It seems intentional that it is not.

I thought the group by was already "in the count", b/c if you say `numberOfInlineFilters=4`, but are also using a group by, then the number of "filter slots" drops to 3.

Fwiw for that job log screenshot, this is nor your fault, but fwiw if the designer had stuck to a more traditional blueprint page layout, of having filters not in the page header, this would not be a problem.

Maybe you or @crshannon could relay that feedback, that we should just move filters out of the page header, b/c a) we're short on space, and b) they are coloring outside of the lines of blueprint established ux patterns?

Reply:
E.g. see specs & selections which puts "Search" and "Filters" closer to the table:

![image](https://user-images.githubusercontent.com/6401/206284466-b95cc777-c890-4112-a3a9-a918fcdab2b8.png)

Granted, I get "this is a list of cards" and not a table, but :shrug: .

Reply:
@stephenh 

I think this image captures the group by not in the more filters better:
![Screen Shot 2022-12-07 at 12 08 57 PM](https://user-images.githubusercontent.com/100221471/206534632-0481f51b-835d-4487-a607-806a141b8cb3.png)

the control here is set to 0 and the group by still exist outside the more filters.

after discussion with the team, design is going to work on this page. We would also like to get this change in so we can alleviate some issues users are currently seeing. 

thank you for the feed back :)

Reply:
@aTrujilloHomeBound I'm unclear what you're trying to ask/communicate with the "group by not in the more filters". 

Naively, wouldn't setting `numberOfInlineFilters=1` and then using a `groupBy` achieve what is in the screenshot? E.g. we've shoved all filters into More Filters. Are you saying "it's good this is now possible" or "there is still something wrong with this"?

That confusion aside, with the `numberOfInlineFilters = groupBy ? 3 : 4` feedback incorporated, yeah I'm good shipping this PR.

Reply:
@stephenh ,

> Naively, wouldn't setting numberOfInlineFilters=1 and then using a groupBy achieve what is in the screenshot? E.g. we've shoved all filters into More Filters. Are you saying "it's good this is now possible" or "there is still something wrong with this"?

With the `numberOfinlineFilters=1` we would expect to see the behavior above, however this is done with the `numberOfInlineFilters=0`, I was unable to find an effective way to get the groupby into the morefilters modal.

Which is why it seemed to me, intentional the groupby exist outside the more filters.

Reply:
> effective way to get the groupby into the morefilters modal

Oh, you want even the group by itself to be moved into the "More Filters" modal?

Reply:
If that is possible that may be a request in the future, 

right now the ball has been passed back to design. However I could see the groupby not being the most vital thing to show in a smaller layout